### PR TITLE
[BACKLOG-41134] [BACKLOG-40475] Disable Folder Actions under Repository 

### PR DESCRIPTION
**Pull request by singletonc**
- Disable all Folder Actions for root folders "Repository" and "VFS Connections"
- Hide icons for root folders, to communicate to users that they shouldn't be considered standard folders

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 22s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5638/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5638/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js [hunk 1]
```diff
@@ -2029,6 +2040,10 @@ define([
     return path.charAt(0) === REPOSITORY_ROOT_PATH;
   }
 
+  function isRootPath(path) {
+    return isRepositoryRootPath(path) || path === "pvfs://";
```
Comment (dcleao) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:2044]: This is a simplification of the concept of generic files and generic file paths.
In general, and even though this is not currently used in the PUC side, these could include non-pvfs URLs, such as `recents://`.

Checking directly if a path is a root path, would require checking for the structure `<scheme>://`. Ideally, we'd be using a shared library to hold this logic, but we don't currently have one we could use for plain JS. The closest are the methods in the `Encoder` module.

Another way would be to use the information on the `GenericFile` data structure returned by the endpoint. A root path is a file whose `parentPath` is `null`.

Also, either case, I think it would be better to call these "provider roots". Keeping this method, it would be called `isProviderRootPath(.)`.

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js [hunk 2]
```diff
@@ -716,10 +719,14 @@ define([
...
 
+          if( isRootPath(obj.file.path) ){
+            obj.file.isRootPath = true;
+          }
+
           //decode potentially encoded name and title attributes of PVFS paths
           if( !isRepositoryPath(obj.file.path) && !obj.file.decoded ){
             obj.file.name = this.decodePvfsFileAttribute(obj.file.name);
-            obj.file.title = obj.file.title === null ? null : this.decodePvfsFileAttribute(obj.file.title);
+            obj.file.title = obj.file.title === null ? obj.file.name : this.decodePvfsFileAttribute(obj.file.title);
```
Comment (dcleao) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:729]: Does this belong to this PR? Is it for BACKLOG-41091?

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js [hunk 3]
```diff
@@ -91,10 +91,10 @@ define([
           "{{#ifCond file.path '.trash'}}" +
           "<div id='{{file.objectId}}' class='trash folder' path='{{file.path}}' ext='{{file.name}}' desc='{{file.name}}'>" +
           "{{else}}" +
-              "{{#if file.title}}" +
-                  "<div id='{{file.objectId}}' class='folder' path='{{file.path}}' desc='{{file.description}}' ext='{{file.name}}' title='{{file.title}}'>" +
+              "{{#if file.isRootPath}}" +
+                  "<div id='{{file.objectId}}' class='folder rootFolder' path='{{file.path}}' desc='{{file.description}}' ext='{{file.name}}' title='{{file.title}}'>" +
               "{{else}}" +
-                  "<div id='{{file.objectId}}' class='folder' path='{{file.path}}' desc='{{file.description}}' ext='{{file.name}}' title='{{file.name}}'>" +
+                  "<div id='{{file.objectId}}' class='folder' path='{{file.path}}' desc='{{file.description}}' ext='{{file.name}}' title='{{file.title}}'>" +
```
Comment (dcleao) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js:97]: Does this belong to this PR? Is it for BACKLOG-41091?

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js [hunk 4]
```diff
@@ -368,18 +368,21 @@ define([
     },
 
     updateFolderButtons: function( _folderPath) {
-      const isRepoPath = !!_folderPath && isRepositoryPath(_folderPath);
+      const isRepositoryFolderPath = !!_folderPath && !isRepositoryRootPath(_folderPath) && isRepositoryPath(_folderPath);
```
Comment (dcleao) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:371]: I think it would be better (at least easier to read) to test for `isRepositoryPath(.)` before `!isRepositoryRootPath(.)`.

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java [hunk 5]
```diff
@@ -413,6 +413,10 @@ public boolean isSupportedExtension( String extension ) {
     return supportedFileExtensions.contains( extension );
   }
 
+  public static List<String> getSupportedGenericFileExtensions() {
```
Comment (dcleao) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java:416]: Does not appear related to these stories.

**Review by dcleao — CHANGES_REQUESTED** [review]

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java (hunk 5 above)
Comment (singletonc) [reply user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java:416]: I will fix this along with https://github.com/pentaho/pentaho-platform/pull/5638#discussion_r1644574159 and do it in my upcoming PR with UTs and other small changes (including VFS File supported file extension on the server side)

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js (hunk 3 above)
Comment (singletonc) [reply user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js:97]: I did a small optimization here... I had been handling "title is null" logic in the template, but that should be handled in a different PR

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js (hunk 2 above)
Comment (singletonc) [reply user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:729]: Related to https://github.com/pentaho/pentaho-platform/pull/5638#discussion_r1644574159, I will move this to another PR

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js (hunk 1 above)
Comment (singletonc) [reply user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:2044]: Closing this PR, I addressed this comment in https://github.com/pentaho/pentaho-platform/pull/5641

**Comment by singletonc** [comment]
Closing PR for new, focused PR from new branch https://github.com/pentaho/pentaho-platform/pull/5641
